dts/arm/atmel/sam3x.dtsi

```diff
@@ -15,6 +15,7 @@
 		cpu@0 {
 			device_type = "cpu";
 			compatible = "arm,cortex-m4";
+			clock-frequency = <84000000>;
```

Comment:
I believe we shouldn't define clock frequency in the SoC dtsi file. This is board and application specific. Even as a default value. We don't do it for other parameters. The `clock-frequency` should be defined in the board .dts file.

Reply:
For boards based on the same SoC and having different clock frequency, the frequency is specified in board file. Otherwise I put the defaults in the SoC file.

If you think that this should be moved, free free to do so.

Reply:
I understand that this PR was a lot of work and what I'm asking for is still more. However, until now we didn't put any board specific configuration parameters in the SoC .dtsi files. Your PR breaks this policy.

The possible clock frequency that the CPU can run with depends strictly on the quartz mounted on the PCB. This is board specific. Even SoCs with PLL, like this one, are limited by the choice of quartz. There is really no default clock frequency value, it either corresponds to the PCB or it is wrong. We can't define CPU frequency here just because some evaluation board happens to be shipped with some specific quartz and not the other.

What we could put in SoC .dtsi files is CPU frequency that can be configured with the built in RC oscillator. So any board could boot safely even if the board .dts file didn't provide clock-frequency property. But we still would need to touch all the .dts files of boards that use quartz and define clock-frequency property there.

Since we want to add clock-frequency property using .yaml files in a separate PR. I can help you by updating all the device drivers, SoC and board files for SiLabs and Atmel. When the other PR is merged.

Reply:
@mnkp: I did what you requested. Please re-review.